src/Microsoft.OData.Core/Json/JsonValueUtilsAsync.cs

```diff
@@ -490,5 +494,136 @@ private static Task WriteQuotedAsync(this TextWriter writer, string text)
             return writer.WriteAsync(
                 string.Concat(JsonConstants.QuoteCharacter, text, JsonConstants.QuoteCharacter));
         }
+
+        /// <summary>
+        /// Writes an escaped string to the buffer.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output.</param>
+
+        /// <param name="inputString">Input string value.</param>
+        /// <param name="currentIndex">The index in the string at which copying should begin.</param>
+        /// <param name="buffer">Char buffer to use for streaming data.</param>
+        /// <param name="bufferIndex">Current position in the buffer after the string has been written.</param>
+        /// <param name="stringEscapeOption">The string escape option.</param>
+        /// <remarks>
+        /// IMPORTANT: After all characters have been written,
+        /// caller is responsible for writing the final buffer contents to the writer.

```

Comment:
how can we guarantee it?

Reply:
It's implemented by the `WriteEscapedJsonStringValueAsync` method: https://github.com/OData/odata.net/pull/2707/files#diff-44795ee4fde32bffe0ac82d8b21b2ac93180517dce7d66dfeb7afce151a1fb8bR447